pgmuvi/lightcurve.py

```diff
@@ -173,49 +173,37 @@ def fit(self, model = None, likelihood = None, num_mixtures = 4,
-        elif model == "1D":
-            self.model = SpectralMixtureGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
-        elif model == "1DLinear":
-            self.model = SpectralMixtureLinearMeanGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
-        elif model == "2DLinear":
-            self.model = TwoDSpectralMixtureLinearMeanGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
-        elif model == "1DSKI":
-            self.model = SpectralMixtureKISSGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
-        elif model == "2DSKI":
-            self.model = TwoDSpectralMixtureKISSGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
-        elif model == "1DLinearSKI":
-            self.model = SpectralMixtureLinearMeanKISSGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
-        elif model == "2DLinearSKI":
-            self.model = TwoDSpectrakMixtureLinearMeanKISSGPModel(self._xdata_transformed,
-                                                    self._ydata_transformed,
-                                                    self.likelihood,
-                                                    num_mixtures = num_mixtures)
+        model_dic_1 = {
+        "2D": TwoDSpectralMixtureGPModel,
+        "1D": SpectralMixtureGPModel,
+        "1DLinear": SpectralMixtureLinearMeanGPModel,
+        "2DLinear": TwoDSpectralMixtureLinearMeanGPModel
+        }
+
+        model_dic_2={
+        "1DSKI": SpectralMixtureKISSGPModel,
+        "2DSKI": TwoDSpectralMixtureKISSGPModel,
+        "1DLinearSKI": SpectralMixtureLinearMeanKISSGPModel,
+        "2DLinearSKI": TwoDSpectrakMixtureLinearMeanKISSGPModel
+        }
+
+        if model is None:
+            pass
```

Comment:
Please replace this with some sort of sensible exception and error message. Maybe ValueError() would make sense?